bw2data/revisions.py

```diff
@@ -36,6 +38,32 @@ def _id(revision: Revision) -> ID:
     return revision["metadata"]["revision"]
 
 
+def _parent(revision: Revision) -> Optional[ID]:
+    return revision["metadata"].get("parent_revision")
+
+
+def _last(x: Iterable[T]) -> T:
+    """Returns the last element of a (non-empty) series."""
+    i = iter(x)
+    ret = next(i)
+    for ret in i:
+        pass
+    return ret
```

Comment:
```suggestion
    return collections.deque(iter(x), maxlen=1).pop()
```

I find this slightly more legible, and is implemented in C. But no strong preferences.

Reply:
I prefer the purely linguistic solution since it's simple, probably equally efficient (none is probably amazing), and exposes pre-condition violations very plainly. Even if Python syntax in this area is not great.

Reply:
Fine by me